code/modules/organs/external/_external.dm

```diff
@@ -108,26 +108,17 @@
 		pain_disability_threshold = (max_damage * 0.75)	
 
 /obj/item/organ/external/Destroy()
-
-	if(wounds)
-		for(var/datum/wound/wound in wounds)
-			qdel(wound)
-
-	if(parent)
-		LAZYREMOVE(parent.children, src)
-		parent = null
-	QDEL_NULL_LIST(children)
-	QDEL_NULL_LIST(internal_organs)
-
+	//Update the hierarchy BEFORE clearing all the vars and refs
+	. = ..()
+	//Clear all leftover refs
+	splinted = null //Splints got deleted in parent proc
+	parent = null
```

Comment:
Unless the logic has been moved somewhere, `parent.children` is not being properly updated.

Reply:
/obj/item/organ/Destroy() calls do_uninstall when stored inside a limb. So no need to do that here.